### PR TITLE
21 define service interfaces

### LINKED ISSUE #21: Define service interfaces
Define service interfaces for:

- [x] Category #10 
- [x] Activity #11 
- [x] Statistics #12

### PR DESCRIPTION
**Pull request by walli545**
Closes #21

### CODE REVIEW

**Review by sschoetz — APPROVED** [review]

File: ctt-frontend/src/app/services/activity.service.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+   * Update an existing activity
+   * 
+   * @param activity, the activity to update.
+   */
+  update(activity: Activity): Promise<void>;
+
+  /**
+   * Deletes an existing activity.
+   * 
+   * @param activity, the activity to create.
```
Comment (sschoetz) [inline ctt-frontend/src/app/services/activity.service.ts]:
```suggestion
   * @param activity, the activity to delete.
```

File: ctt-frontend/src/app/services/activity.service.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  /**
+   * Get all activities relating a category.
+   * 
+   * @param category, the category to match.
+   */
+  getByCategory(category: Category): Promise<Activity[]>;
+
+  /**
+   * Get all activities between from and to. 
+   * And filtilter the cativities relating to the category,
```
Comment (sschoetz) [inline ctt-frontend/src/app/services/activity.service.ts]:
```suggestion
   * And filter the activities relating to the category, 
```

File: ctt-frontend/src/app/services/activity.service.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+   * 
+   * @param from, the included start time.
+   * @param to, the excluded end time.
+   * @param category, the optional category, to filter the activities. 
+   */
+  getBetween(from: Date, to: Date, category?: Category): Promise<Activity[]>;
+
+  /**
+   * Subscribe to all activities between from and to.
+   * If neighter from nor to is provided,
```
Comment (sschoetz) [inline ctt-frontend/src/app/services/activity.service.ts]:
```suggestion
   * If neither from nor to is provided, 
```

File: ctt-frontend/src/app/services/category.service.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+export interface CategoryService {
+  /**
+   * Create a new category.
+   * 
+   * @param input, the category to create.
+   */
+  create(input: CreateCategoryInput): Promise<void>;
+
+  /**
+   * Get an existing category ba an id.
```
Comment (sschoetz) [inline ctt-frontend/src/app/services/category.service.ts]:
```suggestion
   * Get an existing category by an id.
```

File: ctt-frontend/src/app/services/statistics.service.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+   * @param durationTyp, the duration to calculate the distibution.
+   * @param durationValue, the count of the discrete durations to analyze.
+   */
+  getAbsoluteCategoryTime(
+    durationType: QueryDuration,
+    durationValue: number
+  ): Promise<Map<string, number>>;
+
+  /**
+   * Get the categoryID to the absolute
```
Comment (sschoetz) [inline ctt-frontend/src/app/services/statistics.service.ts]:
```suggestion
   * Get the categoryID to the relative
```

**Review by sschoetz — CHANGES_REQUESTED** [review]
Please review suggestions.

**Review by AndHager — APPROVED** [review]

**Review by electron2302 — APPROVED** [review]

**Review by electron2302 — APPROVED** [review]

**Pull request merged**